crates/matrix-sdk-crypto/src/olm/account.rs

```diff
@@ -1511,6 +1513,34 @@ impl PartialEq for Account {
     }
 }
 
+/// Calculate the shared history flag from the history visibility as defined in
+/// [MSC3061]
+///
+/// The MSC defines that the shared history flag should be set to true when the
+/// history visibility setting is set to `shared` or `world_readable`:
+///
+/// > A room key is flagged as having been used for shared history when it was
+/// > used to encrypt a message while the room's history visibility setting
+/// > was set to world_readable or shared.
+///
+/// In all other cases, even if we encounter a custom history visibility, we
+/// should return false:
+///
+/// > If the client does not have an m.room.history_visibility state event for
+/// > the room, or its value is not understood, the client should treat it as if
+/// > its value is joined for the purposes of determining whether the key is
+/// > used for shared history.
+///
+/// [MSC3061]: https://github.com/matrix-org/matrix-spec-proposals/pull/3061
+pub(crate) fn shared_history_from_history_visibility(
+    history_visibility: &HistoryVisibility,
+) -> bool {
+    match history_visibility {
+        HistoryVisibility::Shared | HistoryVisibility::WorldReadable => true,
+        HistoryVisibility::Invited | HistoryVisibility::Joined | _ => false,
+    }
```

Comment:
Since you're using a `_` here, I don't see the point of using a full `match` instead of a simpler:

```rust
matches!(history_visibility, HistoryVisibility::Shared | HistoryVisibility::WorldReadable);
```

Thoughts?

Reply:
The point is to list all of the known ones so people reading the code don't have to run to the spec to see which ones are left.

But yeah, not that much of a point since the enum is non-exhaustive. I'll leave it as is, the extra verbosity doesn't really hurt.

Reply:
Yup your argument is fine to me, thanks.